src/main/java/bio/terra/service/load/LoadDao.java

```diff
@@ -2,33 +2,53 @@
 
 
 import bio.terra.common.DaoKeyHolder;
+import bio.terra.model.BulkLoadFileModel;
+import bio.terra.model.BulkLoadFileState;
+import bio.terra.service.configuration.ConfigEnum;
+import bio.terra.service.configuration.ConfigurationService;
 import bio.terra.service.load.exception.LoadLockFailureException;
 import bio.terra.service.load.exception.LoadLockedException;
 import org.apache.commons.codec.binary.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.CannotSerializeTransactionException;
 import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.BatchPreparedStatementSetter;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Isolation;
 import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.TimeUnit;
 
 @Repository
 public class LoadDao {
-    private final Logger logger = LoggerFactory.getLogger("bio.terra.service.snapshot.LoadDao");
+    private final Logger logger = LoggerFactory.getLogger(LoadDao.class);
 
-    private final NamedParameterJdbcTemplate jdbcTemplate;
+    private NamedParameterJdbcTemplate jdbcTemplate;
+    private ConfigurationService configService;
 
     @Autowired
-    public LoadDao(NamedParameterJdbcTemplate jdbcTemplate) {
+    public LoadDao(NamedParameterJdbcTemplate jdbcTemplate, ConfigurationService configService) {
         this.jdbcTemplate = jdbcTemplate;
+        this.configService = configService;
     }
 
-    @Transactional(propagation = Propagation.REQUIRED)
+    // -- load tags public methods --
+
+    // This must be serializable so that conflicting updates of the locked state and flightid
+    // are detected. The default isolation level for Postgres is READ_COMMITTED; that will allow
+    // the second updater to overwrite the first updater's data. Serialization means the second
+    // updater will throw a PSQLException.
```

Comment:
Thanks for the detailed comment. Just to make sure I follow: The second updater would be another job trying to do the same bulk load, identified by having the same load tag. This could happen if someone called the endpoint twice, before the first call had returned either successfully or with errors. Is that correct?